File: FlowersHub/FlowersHub.Model/Flower.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace FlowersHub.Model
{
    public class Flower
    {
        [Key]
        public string Url { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Currency { get; set; }
        public double PriceDouble { get; set; }
        public string ImageUrl { get; set; }
        public string Group { get; set; }
        public string Updater { get; set; }
        public List<FlowerType> FlowerTypes { get; set; }
        public List<ColorType> Colors { get; set; }

        public Flower()
        {
            FlowerTypes = new List<FlowerType>();
            Colors = new List<ColorType>();
        }

        public Flower(string url, string updater)
        {
            Url = url;
            Updater = updater;
            FlowerTypes = new List<FlowerType>();
            Colors = new List<ColorType>();
        }
    }
}

File: FlowersHub/FlowersHub/clientApp/src/app/Model/flower.ts
export class Flower {
  url?: string;
  title?: string;
  description?: string;
  currency?: string;
  priceDouble?: number;
  imageUrl?: string;
  group?: string;

  constructor(data: any) {
    this.url = (data && data.url) ? data.url : null;
    this.title = (data && data.title) ? data.title : null;
    this.description = (data && data.description) ? data.description : null;
    this.currency = (data && data.currency) ? data.currency : null;
    this.priceDouble = (data && data.priceDouble) ? data.riceDouble : null;
    this.imageUrl = (data && data.imageUrl) ? data.imageUrl : null;
    this.group = (data && data.group) ? data.group : null;
  }
}
